llvm/lib/Support/BlockFrequency.cpp

```diff
@@ -36,3 +37,10 @@ BlockFrequency BlockFrequency::operator/(BranchProbability Prob) const {
   Freq /= Prob;
   return Freq;
 }
+
+bool BlockFrequency::mul(uint64_t Factor, BlockFrequency *Result) const {
```

Comment:
This comment might a bit nitpicky, but how about an interface like this?

```
std::optional<BlockFrequency> BlockFrequency::mul(uint64_t Factor) const {
```

This way, the usage might look like:

```
auto Limit = BBFreq.mul(FreqRatioToSkipMerge);
return Limit && PredFreq <= *Limit;
```

Returning a single result is a little more intuitive to me.  Plus, I don't have to remember whether `true` indicates an overflow or a successful multiplication.

Reply:
Good suggestion!